### PR TITLE
Backend: Improves performance of item backgrounds and borders

### PR DESCRIPTION
**Pull request by ThatGravyBoat**
## What
This PR changes how item backgrounds are handled, previously many locations in the code used the `ItemRenderBackground` class to set the background this now is done individually as to not keep a map of itemstacks and to not move functionality to other parts of the codebase.

## Changelog Technical Details
+ Removed `ItemRenderBackground`. - ThatGravyBoat
    * Added helpers in `RenderUtils` for drawing borders.
    * Added helpers in `RenderUtils` for drawing borders and highlights on `RenderGuiItemOverlayEvent`.
+ Changed `RenderRealOverlayEvent` to `RenderGuiItemOverlayEvent` to be more representative of what the event is for. - ThatGravyBoat

### CODE REVIEW

**Review by CalMWolfs — APPROVED** [review]
worked for me

File: src/main/java/at/hannibal2/skyhanni/features/itemabilities/abilitycooldown/ItemAbilityCooldown.kt [hunk 1]
```diff
@@ -303,16 +307,31 @@ class ItemAbilityCooldown {
...
-                    opacity = 80
-                    if (!config.itemAbilityShowWhenReady) return
-                }
-                stack.background = color.addOpacity(opacity).rgb
+            var opacity = 130
+            if (color == LorenzColor.GREEN) {
+                opacity = 80
+                if (!config.itemAbilityShowWhenReady) return
             }
+            event highlight color.addOpacity(opacity)
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/itemabilities/abilitycooldown/ItemAbilityCooldown.kt:336]: While holding an item with active ability, the whole screen gets a gray effect.
image
image

File: src/main/java/at/hannibal2/skyhanni/features/itemabilities/abilitycooldown/ItemAbilityCooldown.kt [hunk 2]
```diff
@@ -48,7 +49,7 @@ class ItemAbilityCooldown {
     )
 
     private var lastAbility = ""
-    private var items = mapOf<ItemStack, List<ItemText>>()
+    private var items = mapOf<String?, List<ItemText>>()
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/itemabilities/abilitycooldown/ItemAbilityCooldown.kt]: We always check a non-nullable key for the map, the key should not be nullable.

File: src/main/java/at/hannibal2/skyhanni/features/itemabilities/abilitycooldown/ItemAbilityCooldown.kt [hunk 3]
```diff
@@ -254,7 +255,11 @@ class ItemAbilityCooldown {
             abilityItems = ItemUtils.getItemsInInventory(true).associateWith { hasAbility(it) }
         }
 
-        items = abilityItems.mapValues { kp -> kp.value.map { createItemText(it) } }
+        items = abilityItems.entries.associateByTo(
+            mutableMapOf(),
+            { it.key.getIdentifier() },
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/itemabilities/abilitycooldown/ItemAbilityCooldown.kt:261]: filter out when getIdentifier() returns null

**Review by hannibal002 — CHANGES_REQUESTED** [review]
Works fine in my dev env, just the gray effect needs to go.

**Review by hannibal002 — CHANGES_REQUESTED** [review]
while inside inventories, slots with items on them have a different background to other items. this does not happen when pressing E. maybe because of an NEU fancy overlay thing? does not happen in skyblock menu, but in other menus, e.g. bazaar
image

image

**Comment by ThatGravyBoat** [comment]
I cant reproduce, you must have a feature enabled that is conflicting with the highlight changes

**Comment by ThatGravyBoat** [comment]
you have the feature HideNotClickableItems on which does that to show which items you cant click

**Review by hannibal002 — CHANGES_REQUESTED** [review]
<details>
<summary>colours in /tab look desaturated now.</summary>

image

</details>


<details>
<summary>same in inventory</summary>

image

</details>

I think this affects all 35 usages of `infix fun Slot.highlight(color: LorenzColor) {`.

**Comment by ThatGravyBoat** [comment]
Okay fixed this, I by accident removed the lighting when fixing that color setting and attribute changes that were already there that caused the whole screen to have its colored changed.

**Review by hannibal002 — APPROVED** [review]
works fine for me

**Pull request merged**